debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/rest/DebeziumSqlServerConnectorResource.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.sqlserver.rest;
+
+import java.io.IOException;
+import java.util.Map;
+
+import javax.servlet.ServletContext;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.kafka.connect.health.ConnectClusterState;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.debezium.connector.sqlserver.Module;
+
+/**
+ * A JAX-RS Resource class defining endpoints of the Debezium MySQL Connect REST Extension
+ *
+ */
+@Path("/debezium/sqlserver")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class DebeziumSqlServerConnectorResource {
+
+    private final Map<String, ?> config;
+
+    @javax.ws.rs.core.Context
+    private ServletContext context;
+
+    public DebeziumSqlServerConnectorResource(ConnectClusterState clusterState, Map<String, ?> config) {
+        this.config = config;
+    }
+
+    @GET
+    @Path("/version")
```

Comment:
Do you think  `getConnectorVersion` and  `getConnectorSchema` can be moved to an interface with default implementation or to an abstract class and let the implementation to provide just the name of the json?

Reply:
About `getConnectorVersion`: I don't see what this would safe us? We would need to have this method here anyway for the `@Path` annotation and we need to pass the right instance of the `Module` class, which has no interface at all. So this would not work imho. Or do you have another solution?

About `getConnectorSchema`: I was thinking about it earlier too, but I decided against it as we won't safe a lot of duplicate code but we loose independence (the name of that file can be customized on the connector's `pom.xml` level, so if for whatever reason we want to change the name, then we can simply change it in the local implementation. maybe I should even load it from the `Module` class via a System Property eventually so there is only one place to update the name and it is inherited everywhere?) If we would change the name inside the connector's pom.xml it would require us to change the code and add logic for it inside that shared package and release debezium-core (or whatever shared package we use) and with this all connectors that are using that package (even only a single connector might require the change).

Reply:
> About `getConnectorVersion`: I don't see what this would safe us? We would need to have this method here anyway for the `@Path` annotation and we need to pass the right instance of the `Module` class, which has no interface at all. So this would not work imho. Or do you have another solution?

Yes, `Module` is a problem. The only solution I see is to have the `getConnectorVersion` method on the abstract class the calls the abstract method that must be implemented by each connector with the `Module.version()'. 

Regarding the REST annotation this should be feasible (If I am not wrong) https://stackoverflow.com/questions/53667894/jersey-jax-rs-annotating-interface-not-the-implementation-class-results-in-error 

> About `getConnectorSchema`: I was thinking about it earlier too (about having a common abstract class for it), but I decided against it as we won't safe a lot of duplicate code but we loose independence (the name of that file can be customized on the connector's `pom.xml` level, so if for whatever reason we want to change the name or how it is formed, then we can simply change it in the local implementation. maybe I should even load it from the `Module` class via a System Property eventually so there is only one place to update the name and it is inherited everywhere?) If we would change the name inside the connector's pom.xml it would require us to change the code and add logic for it inside that shared package and release debezium-core (or whatever shared package we use) and with this all connectors that are using that package (even only a single connector might require the change).

I was thinking to just push in the abstract class the main code and let the path coming from an abstract method that must be implemented by each connectors. Is it feasible?

Reply:
I am not convinced because we have connectors that are not in the main repo which are required to upgrade when we upgrade the centralized code. I understand the idea of DRY but didn't we also learn that in some distributed environments copying is better than DRY because of lower coupling? And in this case we are talking about 1 line of active code (plus try - catch). Instead of this 1 single line we need that line plus 1 additional method that returns the filename/path in every implementation, plus the coupling introduced to that central abstract implementation that holds the shared code.

Reply:
I have centralized this as an interface now. Please have another look and resolve this discussion if it fits. @mfvitale @jpechane